### PR TITLE
Reduce size of 3.3.2 build without aws bundle

### PR DESCRIPTION
**Pull request by rayhondo**
Use `tabular-client-runtime.jar` instead of `aws-bundle` to minimize size of distribution

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -433,6 +432,11 @@
       <version>1.3.0</version>
       <scope>runtime</scope>
     </dependency>
+    <dependency>
+      <groupId>io.tabular</groupId>
+      <artifactId>tabular-client-runtime</artifactId>
```
Comment (nickwu241) [inline pom.xml:437]: I think in some version of iceberg jar, Iceberg includes everything so we don't even need tabular, can you look at docs or ask Tabular to see if this is true?

Comment (nickwu241) [reply pom.xml:437]: Oh I see the Slack thread at https://affirm-ext.slack.com/archives/C04RZFH2494/p1687827154956799

**Review by nickwu241 — APPROVED** [review]

**Comment by nickwu241** [comment]
What are the sizes now compared to before?

size of before tabular < size with tabular < size with AWS

File: pom.xml (hunk 1 above)
Comment (rayhondo) [reply pom.xml:437]: For Iceberg 1.3.0, its not required, so it should not be needed, but every time I omitted the AWS bundle, it wouldn't work without the `tabular-client-runtime`. TBH - there's a little more to unpack here, but figured I wanted to get something out that was slimmer and then we can further optimize later when needed

**Comment by rayhondo** [comment]
> What are the sizes now compared to before?
> 
> size of before tabular < size with tabular < size with AWS

I didn't check without tabular since it never worked in Alpine or Stage without tabular...

With AWS Bundle - 900 MB
With tabular-client-runtime and without AWS Bundle - 500 MB

**Pull request merged**